src/libraries/System.Net.Security/tests/FunctionalTests/ClientDefaultEncryptionTest.cs

```diff
@@ -17,74 +15,77 @@ public class ClientDefaultEncryptionTest
+                        client.AuthenticateAsClientAsync("localhost", null, SslProtocolSupport.DefaultSslProtocols, false),
+                        server.AuthenticateAsServerAsync(TestConfiguration.ServerCertificate));
+
                     _log.WriteLine("Client authenticated to server({0}) with encryption cipher: {1} {2}-bit strength",
-                        serverAllowNoEncryption.RemoteEndPoint, sslStream.CipherAlgorithm, sslStream.CipherStrength);
-                    Assert.True(sslStream.CipherAlgorithm != CipherAlgorithmType.Null, "Cipher algorithm should not be NULL");
-                    Assert.True(sslStream.CipherStrength > 0, "Cipher strength should be greater than 0");
+                        clientStream.Socket.RemoteEndPoint, client.CipherAlgorithm, client.CipherStrength);
+                    Assert.True(client.CipherAlgorithm != CipherAlgorithmType.Null, "Cipher algorithm should not be NULL");
+                    Assert.True(client.CipherStrength > 0, "Cipher strength should be greater than 0");
                 }
             }
         }
 
         [Fact]
         public async Task ClientDefaultEncryption_ServerNoEncryption_NoConnect()
         {
-            using (var serverNoEncryption = new DummyTcpServer(
-                new IPEndPoint(IPAddress.Loopback, 0), EncryptionPolicy.NoEncryption))
-            using (var client = new TcpClient())
+            (NetworkStream clientStream, NetworkStream serverStream) = TestHelper.GetConnectedTcpStreams();
+            using (clientStream)
+            using (serverStream)
             {
-                await client.ConnectAsync(serverNoEncryption.RemoteEndPoint.Address, serverNoEncryption.RemoteEndPoint.Port);
-
-                using (var sslStream = new SslStream(client.GetStream(), false, AllowAnyServerCertificate, null))
+                using (var client = new SslStream(clientStream, false, TestHelper.AllowAnyServerCertificate, null))
+                using (var server = new SslStream(serverStream, false, TestHelper.AllowAnyServerCertificate, null, EncryptionPolicy.NoEncryption))
                 {
+                    Task serverTask = server.AuthenticateAsServerAsync(TestConfiguration.ServerCertificate);
                     await Assert.ThrowsAsync<AuthenticationException>(() =>
-                        sslStream.AuthenticateAsClientAsync("localhost", null, SslProtocolSupport.DefaultSslProtocols, false));
+                        client.AuthenticateAsClientAsync("localhost", null, SslProtocolSupport.DefaultSslProtocols, false));
+                    try
+                    {
+                        await serverTask.WaitAsync(TestConfiguration.PassingTestTimeout);
+                    }
+                    catch (Exception ex)
+                    {
+                        // serverTask will fail. Log server error in case the test fails.
+                        _log.WriteLine(ex.ToString());
+                    }
```

Comment:
I am not sure how this is supposed to work. If an exception gets thrown in the try block, we swallow it here and the test will pass, if it is thrown elsewhere, then we never get to this catch block.

Same for the other occurrences of this pattern.

Reply:
The original test (and same of the other cases)  only cares about the client behavior. So from functional prospective e.g. test passing or failing only the `Assert.ThrowsAsync` above matters. 

I added the block for `serverTask` for two reasons. One is to collect that tasks before test finishes. And to avoid unobserved task exception. 
The other reasons debugging. _log.WriteLine will take the server exception and it would write to stream provided by xUnit. If the test case fails for what ever reason this would surface in Console.log and testResults.xml. If the test passes there should be no visible impact.

Reply:
I see, then I think the comment should mention that we don't care about the server-side exception in those tests

Reply:
updated.